test/langtools/tools/javac/Diagnostics/compressed/T8012003b.out

```diff
@@ -1,7 +1,7 @@
 T8012003b.java:30:12: compiler.err.prob.found.req: (compiler.misc.invalid.mref: kindname.method, (compiler.misc.cant.apply.symbol: kindname.method, g, java.lang.String, compiler.misc.no.args, kindname.class, T8012003b, (compiler.misc.arg.length.mismatch)))
 T8012003b.java:31:16: compiler.err.prob.found.req: (compiler.misc.stat.expr.expected)
 T8012003b.java:32:22: compiler.err.prob.found.req: (compiler.misc.incompatible.ret.type.in.lambda: (compiler.misc.conditional.target.cant.be.void))
-T8012003b.java:33:12: compiler.err.prob.found.req: (compiler.misc.invalid.mref: kindname.method, (compiler.misc.prob.found.req: (compiler.misc.inconvertible.types: java.lang.Integer, java.lang.String)))
+T8012003b.java:33:12: compiler.err.prob.found.req: (compiler.misc.invalid.mref: kindname.method, (compiler.misc.cant.apply.symbol: kindname.method, g, java.lang.String, java.lang.Integer, kindname.class, T8012003b, (compiler.misc.no.conforming.assignment.exists: (compiler.misc.inconvertible.types: java.lang.Integer, java.lang.String))))
```

Comment:
Seems like this diagnostic is not getting compressed?

Reply:
yep good catch, checking what's going on

Reply:
I have been debugging and analyzing what is happening here, now we are creating more detailed diags compared to the previous code which was basically creating compact diags and they were making it to the corresponding `MethodResolutionContext::candidates` list, so we were selecting a compact diags from the `candidates` list and using a compact diags to issue the error message. Now we are always creating more detailed diags which sometimes link to a compact diags and sometimes don't as we create a lot of diags. This is the reason we are printing a detailed message in this case. One possible solution could be to try to propagate the compact diag, if a detailed diagnostic has one, and we create a new one but I think that if we do this then probably there is no reason to actually create a detailed diagnostic if a compact one is available, which is what the current code is doing.

Reply:
uploaded another iteration as discussed offline, thanks